### PR TITLE
Enha: Don't supress error logs

### LINKED ISSUE #543: How to configure Sentry not to suppress the errors it sees
Hi,

I am starting with Flutter Sentry for Web and it is working but when an error occurs it suppresses the error from being shown on the console. Is there a way to log the error but all it to throw an exception as it would normally do?

Warm regards,
Zambet

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Closes #543 

Introduces and `onError` callback in the `RunZonedGuarded` intergrations, so that we can call `FlutterError.dumpToConsole` when an error is caught by the integration.

This is done this way, as it's the recommended way per [flutter documentation](https://docs.flutter.dev/testing/errors).

`Note: Consider calling [FlutterError.presentError](https://api.flutter.dev/flutter/foundation/FlutterError/presentError.html) from your custom error handler in order to see the logs in the console as well.`

On web this does not yield the same output, but the error name and the StackTrace is there.

Bildschirm­foto 2023-01-23 um 11 03 21

## :bulb: Motivation and Context

When executing code in custom runZonedGuarded, errors on web are not logged to the console anymore.

## :green_heart: How did you test it?

Added unit test.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

Discuss if we need to handle other cases as well.

### CODE REVIEW

**Comment by denrase** [comment]
@marandaneto Not sure if this is the correct approach. Do we only need something like this for `runZonedGuarded` or also for `Isolate.current.addErrorListener`?

File: flutter/lib/src/sentry_flutter.dart [hunk 1]
```diff
@@ -61,6 +62,16 @@ mixin SentryFlutter {
...
 
+    final runZonedGuardedOnError = flutterOptions.platformChecker.isWeb
+        ? (Object error, StackTrace stackTrace) async {
+            final errorDetails = FlutterErrorDetails(
+              exception: error,
+              stack: stackTrace,
+            );
+            FlutterError.dumpErrorToConsole(errorDetails, forceReport: true);
+          }
+        : null;
```
Comment (marandaneto) [inline flutter/lib/src/sentry_flutter.dart]: Let's extract this to a function, and pass the function to the integration.

**Comment by marandaneto** [comment]
@denrase can you check if this causes problems with the `debugPrint` integration?
Enable `enablePrintBreadcrumbs` and check the `DebugPrintIntegration`, I hope this somehow does not go to a recursive loop or something.

**Comment by denrase** [comment]
@marandaneto Not seeing any difference with `enablePrintBreadcrumbs` when running on web and provoking an error thats thrown in `runZonedGuarded`.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**